### DISCUSSION TITLE
2022-06-15 meeting | Finalizing the High-Level Plans

### DISCUSSION THREAD

**Discussion by 7yl4r** [USF-IMARS/chemtax#7] [Meeting Plans, Agendas, Notes]
This is the first of a periodic meeting that has been set up to help push forward development.

We will be showing off:
* the GUI jupyter notebook plan
* @sebastiandig's R code
* plans for development moving forward
    * gh-discussions for meetings notes (+more?)
    * gh-issues for tracking feature requests, bugs, etc
    * TDD development in python (& R?)

The rest of the time will be spent discussing ideas, setting milestones, and reviewing high-level planning documents.

**Comment by 7yl4r** [comment]
## sebastian's R coding efforts
    * gone through 8/22 scripts
    * making notes of how functions are related
    * functions are very interconnected
    * currently hard-coded w/ brokewest dataset
    * next step would be to add user inputs instead of using hardcoded values
    * want to make shiny app
    * harder stuff is non-linear least squares function
        * matlab uses a different implementation than R. not sure how they will differ.
    * variable names are a bit obscure, `ni`, `x`, `b`, etc
        * Simon: variable naming might be clarified in the word document (`CHEMETAX 2 Octave version.docx`)
    * main data script CHEMTAXBROKEWestb

## discussion
* common input errors are:
    * excel file formatting 
    * pigment misnaming
* taxa checklist: these ones are definitely there, these ones might be there (try with & without them).

Codebase could be split into three parts:
* input file checker
* GUI input generator
* CHEMTAX algorithm
* graphing the output

## TODOs:
* Sebastian : put the R code so far to github
* Tylar: start on the GUI (shiny)

**Reply by Karen71** [reply]
This is my first ever github message! Thanks for a great meeting and I am impressed with how much has already been done🥇. See you in July.

**Comment by 7yl4r** [comment]
regarding the separation of code: I would like to keep this repository as the "main" project for tracking and collecting together the various parts once they are completed. While we are developing I would like to keep the various codebases separate. With this in mind: @sebastiandig : please create a repo for the R code you are writing and add a link in the README of this repo. I will do the same for the GUI shiny app.
